src/main/java/org/isf/dicom/manager/SourceFiles.java

```diff
@@ -137,12 +137,26 @@ private void loadDicomDir(FileDicom fileDicom, File sourceFile, int patient) thr
 				}
 				filesLoaded++;
 				dicomLoader.setLoaded(filesLoaded);
-			} else if (!".".equals(value.getName()) && !"..".equals(value.getName())) {
+			} else if (isValidDirectoryName(value.getName())) {
 				loadDicomDir(fileDicom, value, patient);
 			}
 		}
 	}
 
+	private static boolean isValidDirectoryName(String directoryName) {
+		return !isCurrentDirectory(directoryName) && !isParentDirectory(directoryName);
+	}
+
+	private static boolean isCurrentDirectory(String directoryName) {
```

Comment:
IMHO, I think this is too much refactoring.  This and the next `private static` methods are only called once.   The overhead of a method call seems too great when both tests can be added to `isValidDirectoryName` without two method calls.